src/main/java/com/example/search/engine/service/ValidationService.java

```diff
@@ -0,0 +1,24 @@
+package com.example.search.engine.service;
+
+import com.example.search.engine.exception.ValidationException;
+import com.example.search.engine.validator.Validator;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.Assert;
+
+import java.util.List;
+
+
+public class ValidationService implements Validator {
+
+    public void validate(List<String> input) {
+        Assert.notNull(input,"input can not be null");
+        String illegalSymbols = "!@#$%^&*()_+-=";
+        input.stream()
```

Comment:
We expect a very large number of inputs and here we do not care of the order. To increase performance please use parallelStream